src/downloading/dumpFurniFromFurniData.ts

```diff
@@ -9,41 +9,71 @@ export async function dumpFurniFromFurniData(
   folder: string,
   dispatch: (action: Action) => void
 ) {
+  console.time("downloadAndDump");
+
   const data = await parseStringPromise(furniData);
 
   const furniTypes: any[] = data.furnidata.roomitemtypes[0].furnitype;
   const wallTypes: any[] = data.furnidata.wallitemtypes[0].furnitype;
 
   const usedIds = new Set<string>();
-  const allFurni = [...wallTypes, ...furniTypes];
+
+  // We will filter duplicated values to report the right amount of files to be downloaded and dumped.
+  const allFurni: any[] = [...wallTypes, ...furniTypes].reduce(
+    (items, item) => {
+      const name = item["$"].classname.split("*")[0];
+
+      if (usedIds.has(name)) return items;
+
+      usedIds.add(name);
+
+      return [...items, item];
+    },
+    []
+  );
 
   dispatch({ type: "FURNI_ASSETS_COUNT", payload: allFurni.length });
 
   await Bluebird.map(
     allFurni,
-    async (value) => {
+    async (value: any) => {
       const name = value["$"].classname.split("*")[0];
 
-      if (usedIds.has(name)) return;
-      usedIds.add(name);
+      const revision: number | undefined =
+        value.revision != null ? value.revision[0] : undefined;
+
+      try {
+        await downloadFurni(dcrUrl, revision?.toString(), name, folder);
+
+        dispatch({ type: "FURNI_ASSETS_DOWNLOAD_COUNT" });
+      } catch (e) {}
+    },
+    { concurrency: Infinity } // Safe concurrency to avoid 422 from sulake
```

Comment:
Is this `Infinity` intentional? I believe when I ran it the servers instantly blocked me because of it.